### PR TITLE
Remove PaymentInstruments API

### LINKED ISSUE #371: Remove "instrument" concept from PaymentManager
The `PaymentInstruments` [1] concept was originally designed to support multiple credit cards. It adds confusion to payment handlers that are not fundamentally about credit cards. Since Payment Request API is moving towards payment apps being the atomic building blocks instead of instruments, I think it will be more clear to change payment handler registration interface to something like this:

```
registration.paymentManager.setPaymentMethods([
    'https://bobpay.xyz/pay',
    'https://alicepay.com'
]);
```

Today many parts of `PaymentInstruments` are not used:

- Web-based payment handlers usually sets a single instrument with their supported payment methods
- Chrome always fetches the icon from the payment handler's web app manifest instead of using the ones provided in `PaymentInstruments.set()`
- `capabilities` only has well-defined meaning for `basic-card` payment method

With the proposed change to remove the `PaymentInstrument` concept, the web app manifest can become the authoritative source of information about the payment handler, such as name, icons, capabilities. This will also better unify just-in-time installation and explicit installation of payment handlers.

[1] https://w3c.github.io/payment-handler/#paymentinstruments-interface

### LINKED ISSUE #381: Payment Instrument architecture clarification
Hi all,

The BobPay payment method have two payment instruments - bobpay and basic-card. When the user select basic-card, enter needed data and press pay - we use the BobPay's service worker for handling of the rest of the flow (and same for BobPay instrument). Probably it is due to the fact that basic-card is build in instrument.

I want to create a custom Payment Web App with name (for ex.) CustomPay. I want that this app has two payment instruments, first CustomPay and the second NewPay.

**Q:** What should I do, in order to use NewPay service worker when the user select NewPay payment instrument? Could you please give an example? Or all payment logic (for all payment instruments) should be covered with CustomPay service worker? 

Thank you for your answer.
Roman.

### PR DESCRIPTION
**Pull request by rsolomakhin**
PaymentInstruments is the Web API that backs non-JIT install of payment apps. It was designed with the assumption that the browser would store the actual payment instrument details, which has not turned out to be true, and has some privacy leaks. This patch removes the ability to register a payment handler via `paymentManager.instruments` API. Only just-in-time installation is supported from now on.

Chrome tracking bug: https://crbug.com/1327265
Chrome status: https://chromestatus.com/feature/5099285054488576

Overview of changes in the patch:
 * Remove PaymentInstruments interface.
 * Remove no longer used PaymentInstrument and ImageOjbect dictionaries.
 * Mark PaymentManager exposed only on Window (not Worker) to match the Chrome implementation.
 * Add a definition for "payer" for symmetry with "payee" definition.
 * Soften the language around the purpose for the "canmakepayment" event to match Chrome implementation, where it is being used for hasEnrolledInstrument() instead of canMakePayment() and show().
 * Add a promise to the "canmakepayment" example to reduce confusion stemmed from many examples that always unconditionally return "true".
 * Remove the unused PaymentRequestEvent.fetchedImage internal slot.
 * Raise 2nd level sections to 1st level for clarity.

Closes: https://github.com/w3c/payment-handler/issues/371
Closes: https://github.com/w3c/payment-handler/issues/381

***
<a href="https://pr-preview.s3.amazonaws.com/rsolomakhin/payment-handler/pull/409.html" title="Last updated on Jan 25, 2023, 4:13 PM UTC (ff9c715)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-handler/409/6b9aae5...rsolomakhin:ff9c715.html" title="Last updated on Jan 25, 2023, 4:13 PM UTC (ff9c715)">Diff</a>

### CODE REVIEW

**Comment by rsolomakhin** [comment]
@ianbjacobs & @stephenmcgruer: Please take a look.

File: index.html [hunk 1]
```diff
@@ -97,9 +97,8 @@ <h2>
         <li>A payment request event type {{PaymentRequestEvent}}). A <a>payment
```
Comment (stephenmcgruer) [inline index.html]:
```suggestion
        <li>A payment request event type ({{PaymentRequestEvent}}). A <a>payment
```

Nit, while you're here :).

(or alternatively, `A payment request event type, {{PaymentRequestEvent}}.` No closing `)`)

File: index.html [hunk 2]
```diff
@@ -97,9 +97,8 @@ <h2>
...
         handler</a> is an event handler for the {{PaymentRequestEvent}}.
         </li>
-        <li>An extension to the service worker registration interface
-        ({{PaymentManager}} to manage the definition, display, and user
-        selection of {{PaymentInstrument}}s.
+        <li>An extension to the <a>service worker registration</a> interface
+        ({{PaymentManager}} to manage properties of payment handlers.
```
Comment (stephenmcgruer) [inline index.html]:
```suggestion
        ({{PaymentManager}}) to manage properties of payment handlers.
```

Or alternatively, remove the leading `(` and add commas instead

File: index.html [hunk 3]
```diff
@@ -162,21 +143,17 @@ <h2>
...
         </li>
-        <li>When the user (the payer) selects an <a data-lt=
-        "PaymentManager.instruments">instrument</a>, the user agent fires a
-        {{PaymentRequestEvent}} (cf. the <a>user interaction task source</a>)
-        in the service worker whose <a data-lt=
-        "ServiceWorkerRegistration.paymentManager">PaymentManager</a> the
-        instrument was registered with. The {{PaymentRequestEvent}} includes
+        <li>When the <a>payer</a> user selects a payment handler, the user agent
+        fires a {{PaymentRequestEvent}} (cf. the <a>user interaction task
+        source</a>) in its service worker. The {{PaymentRequestEvent}} includes
```
Comment (stephenmcgruer) [inline index.html]: Nit; `its` is ambiguous here (refers to either the payment handler or to the user agent). Needs to be: `in the service worker for the selected payment handler` or similar

File: index.html [hunk 4]
```diff
@@ -289,15 +246,14 @@ <h2>
         </figure>
       </section>
     </section>
+    </section>
```
Comment (stephenmcgruer) [inline index.html:249]: Double-close here? Or are the indentation levels wrong?

**Comment by ianbjacobs** [comment]
@rsolomakhin, one observation is that removing the PaymentInstruments interface should close a couple of other issues, namely #371 and maybe #381.

File: index.html [hunk 5]
```diff
@@ -942,40 +467,39 @@ <h2>
...
       <section>
         <h2>
-          Filtering of Payment Instruments
+          Filtering of Payment Handlers
         </h2>
         <p>
-          Given a <a>PaymentMethodData</a> and a {{PaymentInstrument}} that
-          match on <a>payment method identifier</a>, this algorithm returns
-          <code>true</code> if this instrument can be used for payment:
+          Given a <a>PaymentMethodData</a> and a payment handler that match on
```
Comment (ianbjacobs) [inline index.html]: that "matches" on ...

**Review by ianbjacobs — APPROVED** [review]

File: index.html (hunk 5 above)
Comment (rsolomakhin) [reply index.html]: Done, thank you for noticing this! 😆

**Comment by rsolomakhin** [comment]
> @rsolomakhin, one observation is that removing the PaymentInstruments interface should close a couple of other issues, namely #371 and maybe #381.

Good point! I've added these to the description of the pull request.

**Review by stephenmcgruer — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (rsolomakhin) [reply index.html]: I seem to have a paren match problem 😅

File: index.html (hunk 4 above)
Comment (rsolomakhin) [reply index.html:249]: Indent is wrong. Indent and formatting can be fixed in a separate pull request -- it's lower priority and can confuse reviewers, so I did not fix it here.

File: index.html (hunk 3 above)
Comment (rsolomakhin) [reply index.html]: Done.

**Pull request merged**